### PR TITLE
chore: add pipeline-type config for sdg in standalone

### PR DESCRIPTION
**Pull request by HumairAK**
Depends on: https://github.com/redhat-et/ilab-on-ocp/pull/120

Do not merge until #120 is merged. 
Will need to be rebased after #120 is merged. 

## Description 

This change parameterized this field: https://github.com/instructlab/sdg/blob/59ba8a5bb1fc29642c4e4d79f86e429feefda15f/src/instructlab/sdg/generate_data.py#L282

Keeping `simple` as the default behavior.

### CODE REVIEW

File: standalone/README.md
Comment (leseb) [inline standalone/README.md:621]: Because we use `simple` as the default, I think it's confusing to mark this field as required in the doc. I would treat this as required if we didn't have a default value. What do you think?

File: standalone/README.md
Comment (leseb) [inline standalone/README.md:428]: Typo: "Deafault"

File: standalone/README.md
Comment (HumairAK) [reply standalone/README.md]: yeah I agree, I've added a new commit so you can see the changes, let me know if this works

Comment (leseb) [reply standalone/README.md]: LGTM. Let's just squash everything before we merge :)

File: standalone/standalone.py
Comment (leseb) [inline standalone/standalone.py:59]: Prefix with SDG to stay consistent?

**Review by leseb — COMMENTED** [review]
Can we clarify that this is just for standalone and not the "pipeline"?

File: standalone/standalone.tpl
Comment (leseb) [inline standalone/standalone.tpl:2031]: You also need to account for the 

`elif sdg_serving_model_secret:` 

We must validate that the key `pipeline_type` is present and that the value is either simple or agentic

**Review by leseb — CHANGES_REQUESTED** [review]

File: standalone/standalone.tpl
Comment (HumairAK) [reply standalone/standalone.tpl:2031]: addressed

File: standalone/standalone.py
Comment (leseb) [inline standalone/standalone.py:2893]: This could be a programming preference so feel free to reject it but I'd personally treat both cases separately:

1. the `pipeline_type` key does not exist
2. the key exists but the value if incorrect

With this we can raise more explicit errors.


```python
if not secret.data.get("pipeline_type"):
    raise ValueError("The provided secret must contain a 'pipeline_type' key.")

if secret.data["pipeline_type"] not in ["simple", "agentic"]:
    raise ValueError(
        f"The provided secret {sdg_serving_model_secret} must contain a valid "
        f"value for 'pipeline_type', one of [simple, agentic]."
    )
```

Comment (HumairAK) [reply standalone/standalone.py:2893]: @leseb my understanding from your last comment about making this optional in the secret, means that it is a valid scenario that the this value is not present, so throwing on if not secret.data.get("pipeline_type"): would mean we make this case required, no?

Comment (leseb) [reply standalone/standalone.py:2893]: Actually, you're right forget about what I said, the current approach is correct, if the value is not present we use the default anyway!

**Review by leseb — APPROVED** [review]
LGTM, let's merge once the CI is green

File: standalone/standalone.py
Comment (HumairAK) [inline standalone/standalone.py:544]: may seem awkward placement, but it needs to appear before the @click.option for --sdg-pipeline-type, 
let me know if this should go somewhere else.

File: standalone/standalone.py
Comment (MichaelClifford) [inline standalone/standalone.py:547]: I'm not sure we need this check. How is this handled for other click options? Also, I'm pretty sure `os.path.exists(value):` will fail in all cases since the file you are checking here is what will be called inside of the SDG container not inside the system using the CLI. If this passed it would only be by coincidence.

File: sdg/components.py
Comment (MichaelClifford) [inline sdg/components.py:48]: This seems like it won't work with the standalone CLI (but I assume I'm missing something here). This only allows us to set the `pipeline_type` as an env variable. I assume this will be set correctly when using the `sdg_serving_data_details` secret with the additional 'pipeline_type' variable. But I don't see how the click.options would ever get used here. Does  `ctx.obj["sdg_pipeline_type"]` somehow handle setting it as an env?

File: standalone/standalone.py
Comment (HumairAK) [reply standalone/standalone.py:547]: I don't think other click options are doing this sort of a check. But very good point about the location of the callee, I'll remove it.

File: sdg/components.py
Comment (HumairAK) [reply sdg/components.py:48]: Yes this value is being fed through env vars through the sdg serving secret, just like the endpoint/model/etc. 
https://github.com/redhat-et/ilab-on-ocp/blob/95eebca3bbdb162c3a6ce292f41666aabc3c71d6/standalone/standalone.tpl#L1109

See [here](https://github.com/redhat-et/ilab-on-ocp/pull/133/files#diff-7791f5c215c18f7e52873a5c8da37c76ecf97db053b8122bafc77d570d5ca585R630) for the readme docs

**Review by sallyom — COMMENTED** [review]
I would like to update `pipeline_type` to `sdg_pipeline` and `--sdg-pipeline-type` to `--sdg-pipeline` as long as this works and is consistent with how variables get set in the standalone script, this is good by me.

**Comment by HumairAK** [comment]
we'll split https://github.com/redhat-et/ilab-on-ocp/pull/127/files into 2 PRs, and the 2nd pr will cover the changes here but via a pipeline component arg, not env var

**Pull request closed without merging**
